From 3701b710d27309fbc68325fb345f35f7ec095fa5 Mon Sep 17 00:00:00 2001
From: Micha Kalfon <micha@cellrox.com>
Date: Tue, 13 Jan 2015 12:12:57 +0200
Subject: [PATCH 514/634] prctl: make PR_SET_TIMERSLACK_PID pid namespace aware

Make PR_SET_TIMERSLACK_PID consider pid namespace and resolve the
target pid in the caller's namespace. Otherwise, calls from pid
namespace other than init would fail or affect the wrong task.

Change-Id: I1da15196abc4096536713ce03714e99d2e63820a
Signed-off-by: Micha Kalfon <micha@cellrox.com>
Acked-by: Oren Laadan <orenl@cellrox.com>
---
 kernel/sys.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kernel/sys.c b/kernel/sys.c
index 5414652..4bf3cb0 100644
--- a/kernel/sys.c
+++ b/kernel/sys.c
@@ -2131,11 +2131,11 @@ SYSCALL_DEFINE5(prctl, int, option, unsigned long, arg2, unsigned long, arg3,
 		error = prctl_get_tid_address(me, (int __user **)arg2);
 		break;
 	case PR_SET_TIMERSLACK_PID:
-		if (current->pid != (pid_t)arg3 &&
+		if (task_pid_vnr(current) != (pid_t)arg3 &&
 				!capable(CAP_SYS_NICE))
 			return -EPERM;
 		rcu_read_lock();
-		tsk = find_task_by_pid_ns((pid_t)arg3, &init_pid_ns);
+		tsk = find_task_by_vpid((pid_t)arg3);
 		if (tsk == NULL) {
 			rcu_read_unlock();
 			return -EINVAL;
-- 
1.9.1

